src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -884,69 +879,40 @@ class LocalMemberTable {
                     case CONSTRUCTOR:
                         addMember(e, Kind.CONSTRUCTORS);
                         break;
-                    case ENUM_CONSTANT:
-                        addMember(e, Kind.ENUM_CONSTANTS);
-                        break;
                 }
             }
 
-            // Freeze the data structures
+            // protect from unintended change
             for (Kind kind : Kind.values()) {
-                orderedMembers.computeIfPresent(kind, (k, v) -> Collections.unmodifiableList(v));
-                orderedMembers.computeIfAbsent(kind, t -> List.of());
-
-                memberMap.computeIfPresent(kind, (k, v) -> Collections.unmodifiableMap(v));
-                memberMap.computeIfAbsent(kind, t -> Map.of());
+                orderedMembers.compute(kind, (k, v) -> v == null ? List.of() : Collections.unmodifiableList(v));
+                namedMembers.compute(kind, (k, v) -> v == null ? Map.of() : Collections.unmodifiableMap(v));
```

Comment:
Can just use `replaceAll` than using a for loop

Reply:
There's a cost of changing that to this:

    // protect from unintended change
    orderedMembers.replaceAll((k, l) -> Collections.unmodifiableList(l));
    namedMembers.replaceAll((k, m) -> Collections.unmodifiableMap(m));

That cost is using default values for absent kinds:

    List<Element> getOrderedMembers(Kind kind) {
        return orderedMembers.getOrDefault(kind, List.of());
    }

    List<Element> getMembers(Name simpleName, Kind kind) {
        return namedMembers.getOrDefault(kind, Map.of())
            .getOrDefault(simpleName, List.of());
    }

Or did you mean something else?

Reply:
I don't mind either way; @jonathan-gibbons do you have a preference?

Reply:
TANSTAAFL

I suggest to leave the code as-is.

Reply:
True, my previous suggestion is largely trivial.

In addition, I suggest to run `replaceAll` to ensure the map elements of each value of `namedMembers` is unmodifiable as well:
```
namedMembers.replaceAll((k, m) -> {
    m.replaceAll((n, members) -> Collections.unmodifiableList(members));
    return Collections.unmodifiableMap(m);
});
```

Reply:
I believe that the shallow unmodifiability of namedMembers is a bug, which you spotted. We either should fix it by making it a deeply unmodifiable, or remove this defensiveness altogether: those lists and maps (especially maps) are internal and should never escape. I suppose, it's better to fix it now and revisit later.

Reply:
Or just comment the existing code, indicating that this is just a shallow copy.

Reply:
I think we should follow the original code intent, if that's not too complicated. Please have a look at the proposal in 7565f89, and if you are okay with it, and the tests pass, I'll integrate it.

Reply:
@liach there's no need for namedMembers or its value maps to be unmodifiable: it's the lists that are exposed. So let's protect just them.

Reply:
Sorry, pushed the change sloppily. Since it was done recently and there haven't been more pushes on top of that, I pushed forcefully. Please consider 963bff7 instead of ~7565f89~.